pages/Location/[id].js

```diff
@@ -0,0 +1,18 @@
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import LocationCard from "../../components/LocationCard";
+
+export default function LocationDetail() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const [location, setLocation] = useState(null);
+
+  useEffect(() => {
+    const existingLocations = JSON.parse(localStorage.getItem('favoriteLocations')) || [];
+    const foundLocation = existingLocations.find((loc) => loc.id === id);
+    setLocation(foundLocation);
+  }, [id]);
```

Comment:
The same applies here for the use of local storage: you're better off using the package than manually reading and writing to local storage.